src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/ProcessXmlBase.cs

```diff
@@ -129,9 +129,11 @@ private void ProcessAssembly(ModuleDesc assembly)
             _reader.ReadEndElement();
         }
 
+        protected virtual bool ShouldProcessTypes => true;
+
         private void ProcessType(ModuleDesc assembly)
         {
-            if (ShouldProcessElement())
+            if (ShouldProcessElement() && ShouldProcessTypes)
```

Comment:
```suggestion
            if (ShouldProcessTypes && ShouldProcessElement())
```
`ShouldProcessElement()` looks more expensive compared to the simple bool check of `ShouldProcessTypes` so they should be reversed?